src/main/java/ProductRepo.java

```diff
@@ -0,0 +1,24 @@
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+public class ProductRepo {
+
+    private HashMap<String, Product> products = new HashMap<>();
+
+    public ProductRepo(List<Product> products){
+        for(Product p : products) this.products.putIfAbsent(p.getId(), p);
+    }
+
+    public Optional<Product> get(String id){
+        Optional<Product> productReturn = Optional.of(products.get(id));
```

Comment:
Falls `products.get(id)` null zurückgibt, wirft `Optional.of(products.get(id))` eine `NullPointerException`. Darum würde ich `Optional.ofNullable` verwenden.